### PR TITLE
GODRIVER-1890 Add nil Client checks

### PR DESCRIPTION
**Pull request by benjirewis**
[GODRIVER-1890](https://jira.mongodb.org/browse/GODRIVER-1890)

Adds extra, missing checks for a `nil` `Client` value on an `Operation` to avoid an accidental panic when `Client` is not set on an `Operation`.

### CODE REVIEW

File: x/mongo/driver/operation.go [hunk 1]
```diff
@@ -457,7 +458,9 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 			operationErr.Labels = tt.Labels
 		case Error:
 			if tt.HasErrorLabel(TransientTransactionError) || tt.HasErrorLabel(UnknownTransactionCommitResult) {
-				op.Client.ClearPinnedServer()
+				if op.Client != nil {
```
Comment (iwysiu) [inline x/mongo/driver/operation.go:461]: nit: this one isn't actually necessary, ClearPinnedServer checks if the client it's called on is nil

File: x/mongo/driver/operation.go [hunk 2]
```diff
@@ -634,7 +638,9 @@ func (op Operation) readWireMessage(ctx context.Context, conn Connection, wm []b
 	// everything.
 	op.updateClusterTimes(res)
 	op.updateOperationTime(res)
-	op.Client.UpdateRecoveryToken(bson.Raw(res))
+	if op.Client != nil {
```
Comment (iwysiu) [inline x/mongo/driver/operation.go:641]: same as the other nit

**Review by iwysiu — APPROVED** [review]
lgtm, though as mentioned in comments, some of these checks aren't necessary

**Review by divjotarora — APPROVED** [review]
LGTM with the nits that @iwysiu pointed out.

File: x/mongo/driver/operation.go (hunk 1 above)
Comment (benjirewis) [reply x/mongo/driver/operation.go]: Ah didn't look in the `ClearPinnedServer()` and `UpdateRecoveryToken()` functions. Thank you :) Done.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**